src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_emitter.cpp

```diff
@@ -15,101 +16,37 @@ using namespace dnnl::impl::cpu::x64;
-        leading_dimensions.push_back(repacking_buffer_shape.back());
-    } else {
-        init_in_scheduling_params(input_1_desc);
-    }
-    init_out_scheduling_params(output_desc);
+    m_with_scratch = brgemm_node->is_with_scratchpad();
 
     const auto& brg0Prc = brgemm_node->get_input_element_type(0);
     const auto& brg1Prc = brgemm_node->get_input_element_type(1);
-
-    m_with_scratch = brgemm_node->is_with_scratchpad();
-
-    const auto& output_subtensor = output_desc->get_subtensor();
-    const auto& input_0_subtensor = input_0_desc->get_subtensor();
-    const auto& input_1_subtensor = input_1_desc->get_subtensor();
-
-    OV_CPU_JIT_EMITTER_ASSERT(*(output_subtensor.rbegin() + 1) == *(input_0_subtensor.rbegin() + 1),
-                              "Brgemm has different M dimension subtensors on input0 and output");
-    OV_CPU_JIT_EMITTER_ASSERT(*output_subtensor.rbegin() == *input_1_subtensor.rbegin(),
-                              "Brgemm has different N dimension subtensors on input1 and output");
-    OV_CPU_JIT_EMITTER_ASSERT(*input_0_subtensor.rbegin() == *(input_1_subtensor.rbegin() + 1),
-                              "Brgemm has different K dimension subtensors on input0 and input1");
-
     auto kernel_config = std::make_shared<BrgemmKernelConfig>(brg0Prc, brg1Prc,
                                                             brgemm_node->get_beta(),
                                                             brgemm_node->is_amx(),
                                                             brgemm_node->is_with_compensations());
-
-    m_kernel_executor = kernel_table->register_kernel<BrgemmKernelExecutor>(expr, compiled_kernel_cache, kernel_config);
-    m_kernel_executor->update(*(output_subtensor.rbegin() + 1),
-                              *output_subtensor.rbegin(),
-                              *input_0_subtensor.rbegin(),
-                              leading_dimensions[0],
-                              leading_dimensions[1],
-                              leading_dimensions[2]);
+    std::vector<std::vector<size_t>> layouts;
+    layouts.emplace_back(input_0_desc->get_layout());
+    layouts.emplace_back(input_1_desc->get_layout());
+    layouts.emplace_back(output_desc->get_layout());
+    m_kernel_executor = kernel_table->register_kernel<BrgemmKernelExecutor>(expr,
+                                                                            compiled_kernel_cache,
+                                                                            kernel_config,
+                                                                            layouts);
+    // Note: executor will be updated in runtime if the node is dynamic
+    if (!snippets::utils::is_dynamic_vdims(expr->get_input_port_descriptor(0)->get_shape()) &&
+        !snippets::utils::is_dynamic_vdims(expr->get_input_port_descriptor(1)->get_shape()))
```

Comment:
Yes, we have to trigger the 1st `update()` manually, since the first `ShapeInfer` and `RuntimeConfigurator::update()` are performed before code generation => before registration in the executable kernel table.

An alternative solution is to move the registration into the `RuntimeConfigurator::update()`. Discuss it on review